kratos/containers/model.cpp

```diff
@@ -228,6 +228,57 @@ namespace Kratos
         return nullptr;
     }
 
+    void Model::save(Serializer& rSerializer) const
+    {
+        //we construct auxiliary arrays to avoid having to serialize sets and maps of unique_ptrs
+        std::vector<VariablesList* > aux_var_lists;
+        std::vector<std::string> aux_names;
+        aux_var_lists.reserve(GetListOfVariableLists().size());
+        aux_names.reserve(mRootModelPartMap.size());
+
+        for(auto it = mRootModelPartMap.begin(); it!=mRootModelPartMap.end(); ++it)
+        {
+            aux_names.push_back(it->first);
+        }
+
+        for(auto it = GetListOfVariableLists().begin(); it!=GetListOfVariableLists().end(); ++it)
+            aux_var_lists.push_back(it->get());
+
+        rSerializer.save("ListOfVariablesLists", aux_var_lists);
+        rSerializer.save("ModelPartNames", aux_names);
+
+        for(auto it = mRootModelPartMap.begin(); it!=mRootModelPartMap.end(); ++it)
+        {
+            rSerializer.save(it->first, (it->second).get());
+        }
+    }
+
+    void Model::load(Serializer& rSerializer)
+    {
+        //we construct auxiliary arrays to avoid having to serialize sets and maps of unique_ptrs
+        std::vector<VariablesList* > aux_var_lists;
+        std::vector<std::string> aux_names;
+
+        rSerializer.load("ListOfVariablesLists", aux_var_lists);
+        rSerializer.load("ModelPartNames", aux_names);
+
+        for(IndexType i=0; i<aux_var_lists.size(); ++i) {
+            auto p_aux_list = std::unique_ptr<VariablesList>(aux_var_lists[i]);
+            GetListOfVariableLists().insert(std::move(p_aux_list)); //NOTE: the ordering may be changed since the pointers are changed, however it should not matter
+        }
+
+        for(IndexType i=0; i<aux_names.size(); ++i) {
+            //NOTE: CreateModelPart CANNOT be used here
+            auto dummy_list = Kratos::make_unique<VariablesList>();
+            ModelPart* pmodel_part = new ModelPart(aux_names[i], 1, dummy_list.get(), *this );
```

Comment:
why a raw pointer and not make_unique?

Reply:
this function is using a private constructor. the unique ptr is not a friend of model so it cannot use that, so i first create a raw_pointer and then i assign it to the unique_ptr